### ISSUE TITLE
Set Objects in afterFind and afterLiveQueryEvent triggers

### ISSUE THREAD

**Issue by dblythy** [parse-community/parse-server#7310]
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the [latest version of Parse Server](https://github.com/parse-community/parse-server/releases). 

### Issue Description

With the way Parse handles pointers, if you set a query a Parse.Object and set it to `req.object` using afterFind or afterLiveQueryEvent, Parse converts it to a pointer and strips all fields.

### Steps to reproduce

afterFind:
```js
  it('can set a pointer object in afterFind', async done => {
    //save obj for query
    const obj = new Parse.Object('MyObject');
    await obj.save();
    
    Parse.Cloud.afterFind('MyObject', async ({objects}) => {
      const otherObject = new Parse.Object('Test');
      otherObject.set('foo', 'bar');
      await otherObject.save();
      // set object
      objects[0].set('Pointer', otherObject);
      // set first object field 'Pointer' to object
      expect(objects[0].get('Pointer').get('foo')).toBe('bar');
      return objects;
    });

    const query = new Parse.Query('MyObject');
    query.equalTo('objectId', obj.id);
    const [obj2] = await query.find();
    const pointer = obj2.get('Pointer');
    expect(pointer.get('foo')).toBe('bar');
    // should be able to access 'foo' on the pointer, but it returns null.
    done()

  });
```

afterLiveQueryEvent:

```js
Parse.Cloud.afterLiveQueryEvent('TestObject', async ({object}) => {
  const query = new Parse.Query('Test2');
  const obj = await query.first();
  object.set('obj', obj);
});
subscription.on('update', (object) => {
  expect(object.get('obj')).toBeDefined();
  expect(object.get('obj').get('foo')).toBe('bar');
  // foo will be null
  done();
});
```

### Actual Outcome

Fields on pointer objects are empty

### Expected Outcome

Data should propagate down

### Failing Test Case / Pull Request

- [ ] 🤩 I submitted a PR with a fix and a test case.
- [ ] 🧐 I submitted a PR with a failing test case.

###  Environment

Server
- Parse Server version: `FILL_THIS_OUT`
- Operating system: `FILL_THIS_OUT`
- Local or remote host (AWS, Azure, Google Cloud, Heroku, Digital Ocean, etc): `FILL_THIS_OUT`

Database
- System (MongoDB or Postgres): `FILL_THIS_OUT`
- Database version: `FILL_THIS_OUT`
- Local or remote host (MongoDB Atlas, mLab, AWS, Azure, Google Cloud, etc): `FILL_THIS_OUT`

Client
- SDK (iOS, Android, JavaScript, PHP, Unity, etc): `FILL_THIS_OUT`
- SDK version: `FILL_THIS_OUT`

**Comment by mtrezza** [comment]
Thanks for reporting.

I classified this as a bug. To help determine the severity of this bug:
- Is there a known workaround until this bug is fixed?

**Comment by dblythy** [comment]
Not that I am aware of!

**Comment by dblythy** [comment]
The work around is to fetch the pointer on the frontend.

**Comment by mtrezza** [comment]
Thanks, I classified this as bug with severity S3 (blocks non-critical functionality and a work around exists):
- Workaround: https://github.com/parse-community/parse-server/issues/7310#issuecomment-813020167

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/5.0.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0](https://github.com/parse-community/parse-server/releases/tag/5.0.0)
